### PR TITLE
Use ins singleton pervasively

### PR DESCRIPTION
**Pull request by peterbarker**
We've had an INS singleton for a while now - we just haven't used it all that much.

This PR changes things so that we use the singleton in the libraries.

### CODE REVIEW

**Comment by peterbarker** [comment]
Thanks @lucasdemarchi - that was slack of me!

**Comment by rmackay9** [comment]
LGTM too so I'll leave @OXINARF to do the honours of merging.

**Comment by peterbarker** [comment]
Late patch on top here - I missed the use in `AP_TempCalibration`

**Comment by peterbarker** [comment]
Darn.  @lucasdemarchi Could you force push over this with your changes, please?  I stupidly just lost them :-(

File: libraries/AP_TempCalibration/AP_TempCalibration.h [hunk 1]
```diff
@@ -26,22 +26,27 @@
 class AP_TempCalibration
 {
 public:
-    // constructor
-    AP_TempCalibration(AP_InertialSensor &ins);
+    // constructor.  This remains because construction of Copter's g2
+    // object becomes problematic if we don't have at least one object
+    // to initialise
```
Comment (OXINARF) [inline libraries/AP_TempCalibration/AP_TempCalibration.h]: We don't need to declare the constructor though...

**Comment by peterbarker** [comment]
On Wed, 14 Mar 2018, Francisco Ferreira wrote:

> We don't need to declare the constructor though...

It failed when I didn't :-)

**Comment by peterbarker** [comment]
On Wed, 14 Mar 2018, Francisco Ferreira wrote:

> Regarding the @lucasdemarchi's commits, you can do those yourself... 😃

I couldn't see any way of getting those commits back - I never had them 
locally, and Lucas' github doesn't seem to have a copy....  Yeah, I could 
redo them, but I'm lazy and Lucas has already made the effort :-)

**Comment by OXINARF** [comment]
> It failed when I didn't :-)

Ah probably because we are deleting the copy constructors so it doesn't add the default constructor. You can do `= default;` instead, but whatever.

> I couldn't see any way of getting those commits back - I never had them
locally, and Lucas' github doesn't seem to have a copy....  Yeah, I could
redo them, but I'm lazy and Lucas has already made the effort :-)

Yeah, I meant you doing them! 😄 Not sure if @lucasdemarchi have them, but if he has and commits them in a short timeline, great! Otherwise, I'll pressure you to do it! 😄

**Comment by lucasdemarchi** [comment]
one last rebase and CI run before pushing

**Comment by peterbarker** [comment]
Very, very, very last rebase; had to split commits as @OXINARF pointed out.

This time for sure.

**Pull request merged**